### PR TITLE
Updates custom user-agent header

### PR DESCRIPTION
**Pull request by jmelberg-okta**
All `HTTP` requests are passed a custom header named: `X-Okta-User-Agent-Extended`

### CODE REVIEW

File: Okta/OktaAppAuth.swift [hunk 1]
```diff
@@ -68,7 +68,7 @@ public func clear() {
 }
 
 public func resume(_ url: URL, options: [UIApplicationOpenURLOptionsKey : Any]) -> Bool {
-    if currentAuthorizationFlow!.resumeAuthorizationFlow(with: url){
+    if let authorizationFlow = currentAuthorizationFlow, authorizationFlow.resumeAuthorizationFlow(with: url){
```
Comment (jmelberg-okta) [inline Okta/OktaAppAuth.swift]: Avoid force unwrap

File: Example/Okta/Okta.plist [hunk 2]
```diff
@@ -13,8 +13,8 @@
 	<key>clientSecret</key>
 	<string></string>
 	<key>clientId</key>
-	<string>GJv1mKQtUAUbTalBeQLs</string>
+	<string>{clientId}</string>
 	<key>issuer</key>
-	<string>https://example.oktapreview.com</string>
+	<string>https://j{yourOktaDomain}.com/oauth2/default</string>
```
Comment (marcpowell-okta) [inline Example/Okta/Okta.plist]: intentional prepend of 'j' here?

File: Okta/OktaApi.swift [hunk 3]
```diff
@@ -11,27 +11,53 @@
...
+
         // Generic POST API wrapper
-        
         var request = URLRequest(url: url)
         request.httpMethod = "POST"
-        request.allHTTPHeaderFields = headers
-        
+        request.allHTTPHeaderFields = headers != nil ? headers : request.allHTTPHeaderFields
+        request.addValue(
+            "okta-sdk-appauth-ios/\(VERSION) iOS/\(UIDevice.current.systemVersion)",
```
Comment (bretterer) [inline Okta/OktaApi.swift]: Is there any way to get device info as well as the iOS version?  in the SDK's we have `sdk-name/version language/version os/version`  would it be useful to add to the end of that something like `iPhoneX/blah` if it is available?

File: Okta/OktaApi.swift [hunk 4]
```diff
@@ -11,27 +11,53 @@
...
+    class func get(_ url: URL,
+                   headers: [String: String]?,
+                   callback: @escaping ([String: Any]?, OktaError?) -> Void) {
+
+        // Generic GET API wrapper
+        var request = URLRequest(url: url)
+        request.httpMethod = "GET"
+        request.allHTTPHeaderFields = headers != nil ? headers : request.allHTTPHeaderFields
+        request.addValue(
+            "okta-sdk-appauth-ios/\(VERSION) iOS/\(UIDevice.current.systemVersion)",
```
Comment (bretterer) [inline Okta/OktaApi.swift]: Same as above.

Comment (bretterer) [reply Okta/OktaApi.swift]: Could this also be extracted to a method so we are setting this header in a single place?

Comment (jmelberg-okta) [reply Okta/OktaApi.swift]: Yep - the request logic is going to be modified when we introduce a breaking change to this library. That's coming in a follow up PR.

File: Okta/OktaApi.swift (hunk 3 above)
Comment (jmelberg-okta) [reply Okta/OktaApi.swift]: Done `okta-sdk-appauth-ios/0.3.0 iOS/11.2 Device/x86_64` (Indicates a simulator)

**Pull request merged**